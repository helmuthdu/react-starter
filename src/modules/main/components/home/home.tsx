import React from 'react';
import { FormattedMessage } from 'react-intl';
import { MAIN_ROUTES, USER_ROUTES } from '../../../paths';
import logo from '../../assets/images/logo.svg';

type Props = { onLinkClick: (route: USER_ROUTES | MAIN_ROUTES) => void };
export const Home = (props: Props) => (
  <div className="App">
    <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <p>
        Edit <code>src/app.tsx</code> and save to reload.
      </p>
      <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
        <FormattedMessage id="LEARN_REACT" />
      </a>
      Navigate to
      <div className="App-link" onClick={() => props.onLinkClick(MAIN_ROUTES.ABOUT)} title="go to about page">
        about page
      </div>
      or to
      <div className="App-link" onClick={() => props.onLinkClick(USER_ROUTES.SIGN_IN)} title="go to sign-in page">
        sign-in page
      </div>
    </header>
  </div>
);
