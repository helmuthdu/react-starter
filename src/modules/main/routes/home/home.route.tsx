import { push } from 'connected-react-router';
import React, { Component, HTMLAttributes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, compose, Dispatch } from 'redux';
import { AppState } from '../../../../index';
import { loading } from '../../../../store/modules';
import { AUTH_ROUTES } from '../../../auth/router';
import logo from '../../assets/images/logo.svg';
import { MAIN_ROUTES } from '../../router';

import './home.route.scss';

type State = {};

type StateProps = {
  isLoading: boolean;
};

type DispatchProps = loading.Actions & {
  linkTo: (path: string) => void;
};

type OwnProps = HTMLAttributes<HTMLDivElement>;

export type Props = StateProps & DispatchProps & OwnProps;

export class HomeRoute extends Component<Props, State> {
  async componentDidMount() {
    await new Promise((resolve, reject) => {
      console.log('Before mounting HomePage component', this.props);
      return resolve();
    });
    this.props.toggleLoading();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header" onClick={this.props.toggleLoading}>
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
            Learn React
          </a>
          Navigate to
          <div className="App-link" onClick={() => this.props.linkTo(MAIN_ROUTES.ABOUT)} title="go to about page">
            about page
          </div>
          or to
          <div className="App-link" onClick={() => this.props.linkTo(AUTH_ROUTES.SIGN_IN)} title="go to sign-in page">
            sign-in page
          </div>
        </header>
      </div>
    );
  }
}

const mapStateToProps = (state: AppState): StateProps => ({
  isLoading: loading.getters.isLoading(state)
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps =>
  bindActionCreators(
    {
      linkTo: path => push(path),
      ...loading.actions
    },
    dispatch
  );

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
);

export default enhance(HomeRoute);
