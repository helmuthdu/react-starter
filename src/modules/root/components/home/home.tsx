import React from 'react';
import { USER_URL, ROOT_URL } from '../../../urls';
import logo from '../../assets/images/logo.svg';

type Props = { onLinkClick: (route: USER_URL | ROOT_URL) => void };
export const Home = (props: Props) => (
  <div className="App">
    <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <p>
        Edit <code>src/App.js</code> and save to reload.
      </p>
      <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
        Learn React
      </a>
      Navigate to
      <div className="App-link" onClick={() => props.onLinkClick(ROOT_URL.ABOUT)} title="go to about page">
        about page
      </div>
      or to
      <div className="App-link" onClick={() => props.onLinkClick(USER_URL.SIGN_IN)} title="go to sign-in page">
        sign-in page
      </div>
    </header>
  </div>
);
