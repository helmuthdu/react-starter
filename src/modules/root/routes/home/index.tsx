import * as React from 'react';
import DefaultLayout from '../../../../layouts/default.layout';
import HomeRoute from './home.route';

const Route = () => {
  return (
    <DefaultLayout>
      <HomeRoute />
    </DefaultLayout>
  );
};

export default Route;
