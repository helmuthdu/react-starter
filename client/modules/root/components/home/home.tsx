import Link from 'next/link';
import React from 'react';
import { ROOT_ROUTES, USER_ROUTES } from '../../../routes';

import styles from './home.scss';

type Props = {
  onImageClick: () => void;
};
export const Home = (props: Props) => (
  <div className={styles.app}>
    <header className={styles.appHeader} onClick={props.onImageClick}>
      <img src="/static/assets/images/logo.svg" className={styles.appLogo} alt="logo" />
      <p>
        Edit <code>src/app.js</code> and save to reload.
      </p>
      <a className={styles.appLink} href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
        Learn React
      </a>
      Navigate to
      <Link href={ROOT_ROUTES.ABOUT}>
        <a className={styles.appLink} title="go to about page">
          about page
        </a>
      </Link>
      or to
      <Link href={USER_ROUTES.SIGN_IN}>
        <a className={styles.appLink} title="go to sign-in page">
          sign-in page
        </a>
      </Link>
    </header>
  </div>
);
