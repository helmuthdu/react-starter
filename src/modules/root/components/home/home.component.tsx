import * as React from 'react';
import { USER_ROUTES } from '../../../user/routes/paths';
import logo from '../../assets/images/logo.svg';
import { ROOT_ROUTES } from '../../routes/paths';

type Props = { onImageClick: () => void; onLinkClick: (route: USER_ROUTES | ROOT_ROUTES) => void };

export const Home = (props: Props) => (
  <div className="App">
    <header className="App-header" onClick={props.onImageClick}>
      <img src={logo} className="App-logo" alt="logo" />
      <p>
        Edit <code>src/App.js</code> and save to reload.
      </p>
      <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
        Learn React
      </a>
      Navigate to
      <div className="App-link" onClick={() => props.onLinkClick(ROOT_ROUTES.ABOUT)} title="go to about page">
        about page
      </div>
      or to
      <div className="App-link" onClick={() => props.onLinkClick(USER_ROUTES.SIGN_IN)} title="go to sign-in page">
        sign-in page
      </div>
    </header>
  </div>
);
