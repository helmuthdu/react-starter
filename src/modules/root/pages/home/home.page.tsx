import DefaultLayout from '../../../../layouts/default/default.layout';
import { Home } from '../../components/home/home';

export const HomePage = () => {
  return (
    <DefaultLayout>
      <Home />
    </DefaultLayout>
  );
};

export default HomePage;
