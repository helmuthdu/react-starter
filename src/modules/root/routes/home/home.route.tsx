import * as Sentry from '@sentry/browser';
import { push } from 'connected-react-router';
import React, { Component, ErrorInfo, HTMLAttributes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, compose, Dispatch } from 'redux';
import { AppState } from '../../../../app';
import { loading } from '../../../../stores/modules';
import { Home } from '../../components/home/home.component';
import './home.route.scss';

type StateProps = Readonly<{
  isLoading: boolean;
}>;

type DispatchProps = loading.Actions & {
  linkTo: (path: string) => void;
};

type OwnProps = HTMLAttributes<HTMLDivElement>;

export type Props = StateProps & DispatchProps & OwnProps;

type State = Readonly<{ error: Error }>;

export class HomeRoute extends Component<Props, State> {
  public componentDidMount() {
    this.props.actionToggleLoading();

    return new Promise((resolve, reject) => {
      console.log('Before mounting HomePage component', this.props);
      return resolve();
    });
  }

  public componentDidCatch(error: Error, info: ErrorInfo) {
    this.setState({ error });
    Sentry.captureException(error);
  }

  public render() {
    return <Home onImageClick={this.props.actionToggleLoading} onLinkClick={this.props.linkTo} />;
  }
}

const mapStateToProps = (state: AppState): StateProps => ({
  isLoading: loading.selectors.isLoading(state)
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps =>
  bindActionCreators(
    {
      linkTo: path => push(path),
      ...loading.actions
    },
    dispatch
  );

const enhance = compose<React.FunctionComponent<OwnProps>>(
  connect<StateProps, DispatchProps, OwnProps, AppState>(
    mapStateToProps,
    mapDispatchToProps
  )
);

export default enhance(HomeRoute);
