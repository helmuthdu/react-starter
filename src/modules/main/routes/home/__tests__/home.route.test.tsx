import { shallow } from 'enzyme';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import { HomeRoute, Props } from '../home.route';

describe('Route -> Home', () => {
  const initialState = { ui: { loading: 1 } };

  const props = {
    isLoading: false,
    dispatch: jest.fn(),
    enableLoading: jest.fn(),
    disableLoading: jest.fn(),
    toggleLoading: jest.fn(),
    linkTo: jest.fn(),
    store: configureMockStore()(initialState)
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<HomeRoute {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders welcome message', () => {
    const wrapper = shallow(<HomeRoute {...props} />);
    expect(wrapper.contains('Learn React')).toEqual(true);
  });

  it('should trigger linkTo method', () => {
    const wrapper = shallow(<HomeRoute {...props} />);
    wrapper
      .find('.App-link')
      .at(1)
      .simulate('click');
    expect((wrapper.instance().props as Props).linkTo).toHaveBeenCalled();
  });
});
