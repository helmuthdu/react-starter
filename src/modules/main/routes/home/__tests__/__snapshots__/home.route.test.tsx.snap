// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Route -> Home > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="app-header"
    >
      <img
        alt="logo"
        class="app-logo"
        src="/src/modules/main/assets/images/logo.svg"
      />
      <p>
        Edit 
        <code>
          src/app.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="app-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        LEARN_REACT
      </a>
      Navigate to
      <div
        class="app-link"
        title="go to about page"
      >
        about page
      </div>
      or to
      <div
        class="app-link"
        title="go to sign-in page"
      >
        sign-in page
      </div>
    </header>
  </div>
</DocumentFragment>
`;
